### PR TITLE
Platform: Move RPC code to ocamlformat-rpc library

### PR DESCRIPTION
**Pull request by tmattio**
This moves the code to run the RPC server to the `ocamlformat-rpc` library, allowing users to spawn an ocamlformat RPC server without having to re-write all the logic currently in the executable.

### CODE REVIEW

**Comment by gpetiot** [comment]
I don't have any strong opinion about this, but if we don't lose anything we  can merge. Out of curiosity is there a need from some project?

cc @Julow @panglesd @voodoos @rgrinberg

**Review by gpetiot — APPROVED** [review]
Looks good to me, thanks! Maybe wait for @Julow to also have a look before I merge but it should be good.

File: lib-rpc/dune
Comment (gpetiot) [reply lib-rpc/dune:5]: I also support having 1 library per directory, but to avoid moving the `bin/` files in this PR (and I like having the binaries in `bin/`), maybe it would be enough to move this new library into `lib-rpc-server/` or something similar?

**Review by Julow — APPROVED** [review]
Let's merge.

**Comment by rgrinberg** [comment]
This PR makes ocamlformat-rpc unusable in ocamllsp. ocamllsp requires the most recent version of ocamlformat-rpc-lib, which fixes the constraint for both `ocamlformat` and `ocamlformat-rpc`. The end result is that ocamllsp becomes uninstallable with older versions of ocamlformat.

**Comment by gpetiot** [comment]
@rgrinberg are you sure this is due to this PR and not #2022 ?

**Comment by rgrinberg** [comment]
Ah yes, you're right. I simply picked the first commit in `git blame` and found the corresponding PR.

**Pull request merged**
